kafka-rest/src/main/java/io/confluent/kafkarest/entities/AvroProduceRecord.java

```diff
@@ -70,4 +70,12 @@ public int hashCode() {
     result = 31 * result + (value != null ? value.hashCode() : 0);
     return result;
   }
+
+  public boolean hasKey() {
+    return this.getJsonKey() != null && !this.getJsonKey().isNull();
```

Comment:
It might be a good idea to hoist the result of `getJsonKey()` into a local variable in case that method actually has to do some costly parsing. The same applies to `getJsonValue()` below. JIT might eventually do that for you, but the default threshold for that is still pretty high.

Reply:
it's just a getter and a setter on a locally set variable, there is no parsing.  I could just do key !=null and key.isNull instead, on the already existing key (and value)?